projects/storefrontlib/src/shared/components/form-components/item-counter/item-counter.component.spec.ts

```diff
@@ -47,17 +52,43 @@ describe('ItemCounterComponent', () => {
   }));
 
   beforeEach(() => {
+    isInputFocused = false;
+    isIncrementBtnFocused = false;
+    isDecrementBtnFocused = false;
     fixture = TestBed.createComponent(ItemCounterComponent);
     itemCounterComponent = fixture.componentInstance;
-    itemCounterComponent.input = { nativeElement: { value: '' } };
+    itemCounterComponent.input = {
+      nativeElement: {
+        value: '',
+        focus: () => {
+          isInputFocused = true;
+        },
+      },
+    };
+    itemCounterComponent.incrementBtn = {
+      nativeElement: {
+        value: '',
+        focus: () => {
+          isIncrementBtnFocused = true;
+        },
+      },
+    };
+    itemCounterComponent.decrementBtn = {
+      nativeElement: {
+        value: '',
```

Comment:
`value` is not needed here

Reply:
Thank you for the review and changes you suggested. 
I have fixed all the issues you have brought up.